### PR TITLE
feat: message model, controllers and routes implemented

### PR DESCRIPTION
**Pull request by aaman123**
### What this PR does (required):
- Implementing Messaging backend implemented.

### Screenshots / Videos (required):
- 
Screenshot (1585)

Screenshot (1586)

### Any information needed to test this feature (required):
- Maybe need relevant API calls to test

### Any issues with the current functionality (optional):
- One issue with aggregate mongoose model, will be fixed in the upcoming commit

### CODE REVIEW

**Comment by aaman123** [comment]
@rajivtitus , can u please review this ...

File: server/models/Message.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  messageText: {
+    type: String,
+    required: true,
+  },
+  messageStatus: {
+    type: Boolean,
+    default: false,
+    required: true,
+  },
+  createdAt: {
```
Comment (rajivtitus) [inline server/models/Message.js:24]: Just a suggestion - for future you can consider using mongoose timestamps which will generate this field automatically

Ref: https://masteringjs.io/tutorials/mongoose/timestamps

File: server/controllers/conversation.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        as: "messages",
+      },
+    },
+    {
+      $sort: {
+        "messages.createdAt": 1,
+      },
+    },
+  ]).exec((err, conversations) => {
+    for (var i = 0; i < conversations.length; i++) {
```
Comment (rajivtitus) [inline server/controllers/conversation.js:32]: Please use let instead of var

File: server/controllers/conversation.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+  await Message.find({ conversationId: conversationId }).exec(
+    (err, messages) => {
+      if (err) {
+        res.status(500).send("Something went wrong");
+      } else {
+        res.json({
+          status: true,
+          status_code: 200,
+          conversations: messages,
```
Comment (rajivtitus) [inline server/controllers/conversation.js:59]: Just a typo, conversation will be singular here

File: server/controllers/conversation.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        foreignField: "conversationId",
+        as: "messages",
+      },
+    },
+    {
+      $sort: {
+        "messages.createdAt": 1,
+      },
+    },
+  ]).exec((err, conversations) => {
```
Comment (rajivtitus) [inline server/controllers/conversation.js:31]: Please handle the error here

**Review by rajivtitus — APPROVED** [review]
Good work! Just some minor comments to address before merging

**Pull request merged**